src/test/java/com/techsophy/tsf/wrapperservice/ProcessServiceTest.java

```diff
@@ -414,5 +414,41 @@ void getHistoryTasksByQuery() throws JsonProcessingException {
 
         Assertions.assertEquals(expectedOutput, actualOutput);
     }
+
+    @Test
+    void deleteProcessByIdTest() throws JsonProcessingException {
+        Map<String, Object> map = Map.of("key", "val");
+        ResponseEntity response = Mockito.mock(ResponseEntity.class);
+        Mockito.when(restTemplate.exchange(anyString(), any(), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getBody()).thenReturn(List.of(map));
+        Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.ACCEPTED);
+
+        processService.deleteProcessById("no", "id");
+        verify(restTemplate, times(3)).exchange(anyString(), any(), any(), eq(Object.class));
+    }
+
+    @Test
+    void deleteProcessByIdTestWhileThrowingException() throws JsonProcessingException {
+        Map<String, Object> map = Map.of("key", "val");
+        UriComponentsBuilder uriBuilder = Mockito.mock(UriComponentsBuilder.class);
+        HttpEntity<?> httpEntity = Mockito.mock(HttpEntity.class);
+        ResponseEntity response = Mockito.mock(ResponseEntity.class);
+        Mockito.when(restTemplate.exchange(anyString(), any(), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getBody()).thenReturn(List.of(map));
+        Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.ACCEPTED);
+        Mockito.when(restTemplate.exchange(anyString(), eq(HttpMethod.PATCH), any(), eq(Object.class))).thenThrow(DeleteTaskException.class);
+        Assertions.assertThrows(DeleteTaskException.class, () -> processService.deleteProcessById("no", "id"));
+    }
+
+    @Test
+    void deleteProcessByIdTestWhileTaskNotFound() throws JsonProcessingException {
+        Map<String, Object> map = Mockito.mock(Map.class);
+        List list = List.of(map);
+        ResponseEntity response = Mockito.mock(ResponseEntity.class);
+        Mockito.when(restTemplate.exchange(anyString(), any(), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getBody()).thenReturn(list);
+        Mockito.when(map.get(any())).thenThrow(TaskNotExitsException.class);
```

Comment:
why is this exception thrown from map call.

Reply:
In ProcessServiceImpl, Line no. 433 is
		try {
            processInstanceId = String.valueOf(responseEntity1Body.get(0).get(PROCESS_INSTANCE_ID));
        }
        catch(Exception ex)
        {
            throw new TaskNotExitsException(TASK_NOT_EXIST,UNABLE_TO_FIND_THE_TASK);

        }
responseEntity1Body is a list of map, and while we are fetching the value of map whose key is PROCESS_INSTANCE_ID, then It may throw Exception, That is why in testcase I mock it and throw a Exception.